### PR TITLE
The first step to the 0.2

### PR DESCRIPTION
**Pull request by StannisMod**
This is update about refactorings, fixes and other public version requirements.
- Fixed many (really many) bugs
- Re-read and refactored whole code
- Removed old and unused functions, classes, packages, etc.
- Simplified API of most general functions
- Greatly improved usability
- Simplified (shortened) import paths. For example, `from sampo.scheduler import HEFTScheduler` instead of `from sampo.scheduler.heft.base import HEFTScheduler`
- Added `PreparationPipeline` to fully control of `WorkGraph` and `list[Contractor]` preparation process separately from scheduling
- Made `WorkGraph` fully serializable by `dill` that enables to fully use `pathos multiprocessing` in our project
- Added `Visualization` as a pipeline for visualization things

### CODE REVIEW

File: sampo/schemas/graph.py [hunk 1]
```diff
@@ -277,7 +282,32 @@ def min_start_time(self, node2swork: dict['GraphNode', ScheduledWork]) -> Time:
                     for edge in self.edges_to if edge.start in node2swork), default=Time(0))
 
 
-GraphNodeDict = dict[str, GraphNode]
+def get_start_stage(work_id: str | None = "", rand: Random | None = None) -> GraphNode:
```
Comment (vanoha01) [inline sampo/schemas/graph.py:285]: Work_id is None? Is it necessary mention, if default value is empty string

File: sampo/schemas/graph.py [hunk 2]
```diff
@@ -277,7 +282,32 @@ def min_start_time(self, node2swork: dict['GraphNode', ScheduledWork]) -> Time:
...
+    """
+    Creates a service vertex necessary for constructing the graph of works,
+    which is the only vertex without a parent in the graph
+    :param work_id: id for the start node
+    :param rand: Number generator with a fixed seed, or None for no fixed seed
```
Comment (vanoha01) [inline sampo/schemas/graph.py:290]: Remove capital letter

File: sampo/schemas/graph.py [hunk 3]
```diff
@@ -277,7 +282,32 @@ def min_start_time(self, node2swork: dict['GraphNode', ScheduledWork]) -> Time:
...
+    :return: desired node
+    """
+    work_id = work_id or uuid_str(rand)
+    work = WorkUnit(work_id, 'start of project', [], group='service_works', is_service_unit=True)
+    return GraphNode(work, [])
+
+
+def get_finish_stage(parents: list[GraphNode | tuple[GraphNode, float, EdgeType]], work_id: str | None = "",
```
Comment (vanoha01) [inline sampo/schemas/graph.py:298]: The same as above about None

File: sampo/schemas/graph.py [hunk 4]
```diff
@@ -277,7 +282,32 @@ def min_start_time(self, node2swork: dict['GraphNode', ScheduledWork]) -> Time:
...
+                     rand: Random | None = None) -> GraphNode:
+    """
+    Creates a service vertex necessary for constructing the graph of works,
+    which is the only vertex without children in the graph
+    :param parents: a list of all non-service nodes that do not have children
+    :param work_id: id for the start node
+    :param rand: Number generator with a fixed seed, or None for no fixed seed
```
Comment (vanoha01) [inline sampo/schemas/graph.py:305]: Capital letter (CP)

File: sampo/userinput/parser/general_build.py [hunk 5]
```diff
@@ -213,45 +232,38 @@ def topsort_graph_df(frame: pd.DataFrame) -> pd.DataFrame:
...
+    id_to_node = {}
 
     for _, row in frame.iterrows():
-        if 'min_req' in frame.columns and 'max_req' in frame.columns:
-            reqs = [WorkerReq(res_name, row[res_name],
-                              row['min_req'][res_name],
-                              row['max_req'][res_name]
-                              ) for res_name in resource_names
-                    if 0 < row['min_req'][res_name] <= row['max_req'][res_name]]
+        if 'min_req' in frame.columns and 'max_req' in frame.columns and 'req_volume' in frame.columns:
```
Comment (vanoha01) [inline sampo/userinput/parser/general_build.py:239]: Was list comprehension replaced by going with loop?

**Review by vanoha01 — CHANGES_REQUESTED** [review]
Changed are requested

File: sampo/scheduler/genetic/operators.py [hunk 6]
```diff
@@ -62,7 +63,7 @@ def __init__(self, evaluator: Callable[[list[ChromosomeType]], list[Schedule]],
         self._resources_names = list(resources_names) if resources_names is not None else None
 
     @staticmethod
-    def prepare(resources_names: Iterable[str]):
+    def prepare(resources_names: Iterable[str]) -> Callable[[list[ChromosomeType]], list[Schedule]]:
```
[the remark this answers is not in the document]
Comment (StannisMod) [reply sampo/scheduler/genetic/operators.py:66]: No, check the documentation of `Callable`

File: sampo/scheduler/genetic/operators.py [hunk 7]
```diff
@@ -84,7 +85,7 @@ def __init__(self, evaluator: Callable[[list[ChromosomeType]], list[Schedule]],
...
```
[the remark this answers is not in the document]
Comment (StannisMod) [reply sampo/scheduler/genetic/operators.py:88]: The same as above

**Review by vanoha01 — APPROVED** [review]
Approved

**Pull request merged**